index_test.go

```diff
@@ -306,6 +314,24 @@ func TestPersistence_index_e2e(t *testing.T) {
 		require.NoError(t, gotp.Err())
 	}
 
-	require.NoError(t, ir.Close())
+	for k, v := range mi.labelIndex {
+		tplsExp, err := newStringTuples(v, 1)
+		require.NoError(t, err)
+
+		tplsRes, err := ir.LabelValues(k)
+		require.NoError(t, err)
 
+		require.Equal(t, tplsExp.Len(), tplsRes.Len())
+		for i := 0; i < tplsExp.Len(); i++ {
+			strsExp, err := tplsExp.At(i)
+			require.NoError(t, err)
+
+			strsRes, err := tplsRes.At(i)
+			require.NoError(t, err)
+
+			require.Equal(t, strsExp, strsRes)
+		}
+	}
+
+	require.NoError(t, ir.Close())
```

Comment:
Hmm, removed index.go from the checklist of done things. Will test index once I get some free time. But I don't think this should block this PR.